### PR TITLE
Wildcard support for floating identifiers

### LINKED ISSUE #320: Wildcard support for floating list
Hey,

I'd love to be able to add "unity.*" to my floating list and that would float all apps with an app id starting with unity.

I can even help implementing this, if you can give me pointers as to where I should be looking for it :)

[Trello Card](https://trello.com/c/5dRrFo6h/149-amethyst-wildcard-support-for-floating-list)

### PR DESCRIPTION
**Pull request by deiga**
The changes should work now, still need to add a few tests. And I need comments about the way I solved this :P

Fixes #320 

[Trello Card](https://trello.com/c/7TqJMyKe/199-amethyst-wildcard-support-for-floating-identifiers)

### CODE REVIEW

File: Amethyst/AMConfiguration.m [hunk 1]
```diff
@@ -412,8 +412,20 @@ - (BOOL)runningApplicationShouldFloat:(NSRunningApplication *)runningApplication
     if (!floatingBundleIdentifiers) {
         return NO;
     }
+    for (NSString *floatingBundleIdentifier in floatingBundleIdentifiers) {
```
Comment (ianyh) [inline Amethyst/AMConfiguration.m]: Please use 4 spaces, not 2.

File: Amethyst/AMConfiguration.m [hunk 2]
```diff
@@ -412,8 +412,20 @@ - (BOOL)runningApplicationShouldFloat:(NSRunningApplication *)runningApplication
...
+        }
+      } else {
+        if ([floatingBundleIdentifier isEqualToString:runningApplication.bundleIdentifier]) {
+          return true;
+        }
+      }
+    }
 
-    return [floatingBundleIdentifiers containsObject:runningApplication.bundleIdentifier];
+    return false;
```
Comment (ianyh) [inline Amethyst/AMConfiguration.m]: Should be `NO`.

File: Amethyst/AMConfiguration.m [hunk 3]
```diff
@@ -412,8 +412,20 @@ - (BOOL)runningApplicationShouldFloat:(NSRunningApplication *)runningApplication
     if (!floatingBundleIdentifiers) {
         return NO;
     }
+    for (NSString *floatingBundleIdentifier in floatingBundleIdentifiers) {
+      if ([floatingBundleIdentifier containsString:@"*"]) {
+        NSString *sanitizedIdentifier = [floatingBundleIdentifier stringByReplacingOccurrencesOfString:@"*" withString:@""];
+        if ([runningApplication.bundleIdentifier hasPrefix:sanitizedIdentifier]) {
+          return true;
```
Comment (ianyh) [inline Amethyst/AMConfiguration.m]: `YES` not `true`.

File: Amethyst/AMConfiguration.m [hunk 4]
```diff
@@ -412,8 +412,20 @@ - (BOOL)runningApplicationShouldFloat:(NSRunningApplication *)runningApplication
...
+        }
+      } else {
+        if ([floatingBundleIdentifier isEqualToString:runningApplication.bundleIdentifier]) {
+          return true;
```
Comment (ianyh) [inline Amethyst/AMConfiguration.m]: `YES` not `true`.

**Comment by ianyh** [comment]
The approach is generally okay, but we should consider how this is going to fit into the in-app preferences.

**Comment by deiga** [comment]
You mean in the GUI preferences? Maybe add the option to edit an entered value+

**Comment by deiga** [comment]
Travis failed because of some travis hickups, restarting the build shoud fix it

**Comment by deiga** [comment]
@ianyh I actually don't know where the tests are in the project :P How do I add tests? :)

**Comment by ianyh** [comment]
@deiga I just merged some dependency management changes into `development` if you want to update your branch here.

**Comment by ianyh** [comment]
Thinking about it, it seems pretty reasonable to add a context menu to the plus button for layouts in preferences that gives you the option to either select an application or manually input a value. That is a little involved, though. Do you have much AppKit experience?

**Comment by deiga** [comment]
Not really, no :P

**Comment by ianyh** [comment]
Heh. That's alright. I can take care of the UI stuff after this gets merged. Now that 0.10 is basically in a releasable state I should be able to get this merged in for 0.11 in the next day or two.

**Comment by ianyh** [comment]
0.10 is out so I'm going to merge this!

**Comment by deiga** [comment]
I hope you checked if this works :P

**Pull request merged**